### ISSUE TITLE
Error linting project using wasm-bindgen on nighlty

### ISSUE THREAD

**Issue by alexlapa** [rust-lang/rust-clippy#4507]
`cargo clippy --all -- -D clippy::pedantic -D warnings` is ok

running `cargo +nightly clippy --all -- -D clippy::pedantic -D warnings` produces:

```
error: integer type suffix should be separated by an underscore
  --> jason/src/api/connection.rs:28:1
   |
28 | #[wasm_bindgen]
   | ^^^^^^^^^^^^^^^ help: add an underscore: `#[wasm_bindg_u32`
   |
   = note: `-D clippy::unseparated-literal-suffix` implied by `-D clippy::pedantic`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unseparated_literal_suffix
```

```
cargo clippy -V 
clippy 0.0.212 (e3cb40e 2019-06-25)
```

```
cargo +nightly clippy -V
clippy 0.0.212 (aeadf15 2019-09-03)
```

[Reproduction](https://github.com/alexlapa/rust-clippy-4507-reproduction).

**Comment by flip1995** [comment]
cc #4421 @lzutao 

It seems that
```rust
if let Some(firstch) = src.chars().next();
if char::to_digit(firstch, 10).is_some();
```
was necessary. I think, we should add this back in, instead of the 
https://github.com/rust-lang/rust-clippy/blob/c3d42949c43f10e71e4cd51d1243829522b0b724/clippy_lints/src/misc_early.rs#L434-L439
hack. But this time we should document, why this exists.

**Comment by lzutao** [comment]
I added it back in #4510. But unfortunately, proc-macro is out of my league.
It would be nice if someone could add a regression test for this.

**Comment by flip1995** [comment]
@alexlapa Can you provide a minimal reproducible example or share the code this lint triggered on?

**Comment by alexlapa** [comment]
@flip1995 ,

Sure. https://github.com/alexlapa/rust-clippy-4507-reproduction

**Comment by flip1995** [comment]
Thanks! Extract from expanded code:

```rust
        // ...
        unsafe fn __wbg_foo_new(_: u32) -> u32 {
            {
                ::std::rt::begin_panic("cannot convert to JsValue outside of the wasm target",
                                       &("src/lib.rs", 3u32, 1u32))
            }
        }
        // ...
```

It seems, we only have to write a proc macro which produces code with `0u32` (or similar) in it. This should be easily done by extending
https://github.com/rust-lang/rust-clippy/blob/9672a0400cf5be60e06782f4a1faabade15e8936/mini-macro/src/lib.rs#L7-L8

@lzutao do you want to take a shot at this? Otherwise I can add a test case to the PR this weekend.

**Comment by lzutao** [comment]
Fixed in #4510.
